### DISCUSSION TITLE
GENMON with Briggs and Stratton with Raspberry Pi (HAT)

### DISCUSSION THREAD

**Discussion by esnoop155** [jgyates/genmon#1041] [General]
Hello,
I'm having issues with the communication from GENMON to the Generator.  As I said before, I am connecting with a Raspberry Pi with the serial hat.  I looks like the GENMON is talking to the Generator but not receiving anything back from the Generator. 

I have tried to Change the wires from A to B and Back. I have changed raspberry pi's,  been to setup and changed the parity bit and it would brick the Pi.  I have tried Hardwire and WIFI
![BS](https://github.com/jgyates/genmon/assets/157388037/40ddc492-12c6-422c-8a17-1238383a1126)
![BS1](https://github.com/jgyates/genmon/assets/157388037/da62d1fb-0e93-46cf-8db4-d3dd2a2a54a5)
 no change.

**Comment by jgyates** [comment]
Have you seen this page in the wiki: https://github.com/jgyates/genmon/wiki/Appendix-P-Briggs-and-Stratton-Controller-Information

**Reply by esnoop155** [reply]
Yes we have done everything on this link before posting. But none of this worked to get it connected to the generator.

**Comment by jgyates** [comment]
I am going to close this thread, but feel free to post here with any follow up questions or status.

**Comment by jgyates** [comment]
@esnoop155 
I will need more info to provide any suggestions.

you mentioned " I looks like the GENMON is talking to the Generator but not receiving anything back from the Generator." What serial errors are you getting? Timeout errors, CRC errors? This will be on the Monitor page in the web interface.

This page has info on serial validation. https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting

You mentioned using a HAT. Can you provide pictures of the HAT showing your connections and any jumpers. When using RS485 things like terminating resistors on both ends of the cable can make a big difference.

You mentioned "I have changed raspberry pi's, been to setup and changed the parity bit and it would brick the Pi. I have tried Hardwire and WIFI"

Can you give more detail on "brick the pi"? Does that mean it is not recoverable?, you can not load in the web interface? it will not boot? you can not ping the pi? please give more details.

you also mentioned you tried wifi? I assume this is a serial to wifi adapter. Did you try to validate your comms before you connected to the generator with a loopback using the program ./genmon/OtherApps/sockettest.py. This program is intended to validate your serial to over TCP adapter is working. If you want to use serial over TCP please include the serial converter device you are using and the setup.

Have you validated your pi is setup correctly by performing a loopback test? This page shows how to perform the three stages of loopback (pi only, pi with HAT and Pi with cable). These tests will need to be modified as RS485 is a half duplex communication medium.  

for valiating the serial port, the HAT and the cable we will test without the parity set as we are just concerned with validating the port is working. The serialtest.py will do the test at 9600,N,8,1

Assuming you want to do a direct connection serial, possibly the best way to start is with a simple loopback on the pi without the HAT. If you ran the setup program you should be able to connect pin 8 to 10. use the ./genmon/OtherApps/serialtest.py app to test the loopback. If you are running the 64 bit OS you can type this command to validate your GPIO pins are configured for serial:

     pinctrl get 14,15

Let me know how it goes.

**Reply by jgyates** [reply]
also, if you pi is setup and outbound email is enabled you can submit your logs files from the about page. I can look at the logs and see if anything jumps out as a problem.

**Reply by jgyates** [reply]
Not sure if you saw this or not but there are also a series of videos on using genmon with a Briggs and Stratton generator:

https://www.youtube.com/@genmonbriggs9033/videos

**Reply by esnoop155** [reply]
I will get with my IT guy on the info above that you explained. Also yes we set up everything with the video you sent we initially set it up. So  with the email are you only able to setup Gmail emails only? We tried aol and yahoo with the proper configurations and an error came back each time on the setup.

**Reply by esnoop155** [reply]
Also thank you so much for the quick responses.

**Comment by jgyates** [comment]
Genmon can notify of events via numerous other methods besides email, but if you want ot sent your logs, email is the only methods. Usually with email settings, you have to have an application specific password. This page has several tips for using genmon, including how to setup a gmail account for email.

https://github.com/jgyates/genmon/wiki/Appendix-M-Tips-and-Tricks-useful-for-Genmon

**Reply by esnoop155** [reply]
Great I'll look this over to set the email up
